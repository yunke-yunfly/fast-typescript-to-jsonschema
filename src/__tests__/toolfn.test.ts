import * as path from 'path';

import genTypeSchema from '../index';

const file = path.join(__dirname, `../../jest/toolfn.ts`);
genTypeSchema.genJsonDataFormFile(file);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(file, type);
  return schema;
};

test('工具函数Omit_1', () => {
  expect(getSchema('ToolFn_1')).toMatchSnapshot();
});

test('工具函数Omit_2', () => {
  expect(getSchema('ToolFn_2')).toMatchSnapshot();
});

test('工具函数Omit_3', () => {
  expect(getSchema('ToolFn_3')).toMatchSnapshot();
});

test('工具函数Omit_4', () => {
  expect(getSchema('ToolFn_17_')).toMatchSnapshot();
});

test('工具函数Omit_4', () => {
  expect(getSchema('ToolFn_4')).toMatchSnapshot();
});

test('工具函数Omit_5', () => {
  expect(getSchema('ToolFn_5')).toMatchSnapshot();
});

test('工具函数Omit_6', () => {
  expect(getSchema('ToolFn_6')).toMatchSnapshot();
});

test('工具函数Omit_7', () => {
  expect(getSchema('ToolFn_7')).toMatchSnapshot();
});

test('工具函数Omit_8', () => {
  expect(getSchema('ToolFn_8')).toMatchSnapshot();
});

test('工具函数Pick_1', () => {
  expect(getSchema('ToolFn_9')).toMatchSnapshot();
});

test('工具函数Pick_2', () => {
  expect(getSchema('ToolFn_10')).toMatchSnapshot();
});

test('工具函数Pick_3', () => {
  expect(getSchema('ToolFn_16_')).toMatchSnapshot();
});

test('工具函数Pick_3', () => {
  expect(getSchema('ToolFn_11')).toMatchSnapshot();
});

test('工具函数Pick_4', () => {
  expect(getSchema('ToolFn_12')).toMatchSnapshot();
});

test('工具函数Record_1', () => {
  expect(getSchema('ToolFn_13')).toMatchSnapshot();
});

test('工具函数Record_2', () => {
  expect(getSchema('ToolFn_14')).toMatchSnapshot();
});

test('工具函数Record_3', () => {
  expect(getSchema('ToolFn_15')).toMatchSnapshot();
});

test('工具函数Record_4', () => {
  expect(getSchema('ToolFn_16')).toMatchSnapshot();
});

test('工具函数Record_5', () => {
  expect(getSchema('ToolFn_17')).toMatchSnapshot();
});

// test('工具函数Record_6', () => {
//   expect(getSchema('ToolFn_18')).toMatchSnapshot();
// });


test('工具函数Partial_1', () => {
  expect(getSchema('ToolFn_18_')).toMatchSnapshot();
});

test('工具函数Partial_2', () => {
  expect(getSchema('ToolFn_19')).toMatchSnapshot();
});

test('工具函数Partial_3', () => {
  expect(getSchema('ToolFn_20')).toMatchSnapshot();
});

test('工具函数Partial_联合类型_1', () => {
  expect(getSchema('ToolFn_24')).toMatchSnapshot();
});

test('工具函数Required_1', () => {
  expect(getSchema('ToolFn_21')).toMatchSnapshot();
});

test('工具函数Required_2', () => {
  expect(getSchema('ToolFn_22')).toMatchSnapshot();
});

test('工具函数Required_3', () => {
  expect(getSchema('ToolFn_23')).toMatchSnapshot();
});

test('工具函数Required_联合类型_1', () => {
  expect(getSchema('ToolFn_25')).toMatchSnapshot();
})