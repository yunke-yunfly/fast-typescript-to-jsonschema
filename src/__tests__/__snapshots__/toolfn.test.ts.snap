// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`工具函数Omit_1 1`] = `
Object {
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "c": Object {
      "type": "boolean",
    },
  },
  "required": Array [
    "a",
    "c",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_2 1`] = `
Object {
  "properties": Object {
    "a": Object {
      "type": "number",
    },
  },
  "required": Array [
    "a",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_3 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "c": Object {
      "type": "boolean",
    },
  },
  "required": Array [
    "a",
    "c",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_4 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "c": Object {
      "type": "boolean",
    },
  },
  "required": Array [
    "c",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_4 2`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Omit<{a:number;b:string;c:boolean;},\\"b\\">": Object {
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "a",
        "c",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Omit<{a:number;b:string;c:boolean;},\\"b\\">",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_5 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Omit<{a:number;b:string;c:boolean;},\\"b\\"|\\"c\\">": Object {
      "properties": Object {
        "a": Object {
          "type": "number",
        },
      },
      "required": Array [
        "a",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Omit<{a:number;b:string;c:boolean;},\\"b\\"|\\"c\\">",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_6 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Omit<AAA,\\"b\\"|\\"c\\">": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
      },
      "required": Array [
        "a",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Omit<AAA,\\"b\\"|\\"c\\">",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_7 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "a",
        "b",
        "c",
      ],
      "type": "object",
    },
    "Omit<AAA,\\"b\\"|\\"c\\"_1>": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
      },
      "required": Array [
        "a",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "anyOf": Array [
        Object {
          "$ref": "#/definitions/AAA",
        },
        Object {
          "$ref": "#/definitions/Omit<AAA,\\"b\\"|\\"c\\"_1>",
        },
      ],
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Omit_8 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Omit<AAA,\\"b\\"_0>": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
      },
      "required": Array [
        "a",
      ],
      "type": "object",
    },
    "Omit<AAA,\\"b\\"_1>": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "a",
        "c",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "anyOf": Array [
        Object {
          "items": Object {
            "anyOf": Array [
              Object {
                "$ref": "#/definitions/Omit<AAA,\\"b\\"_0>",
              },
              Object {
                "$ref": "#/definitions/AAA",
              },
            ],
          },
          "type": "array",
        },
        Object {
          "$ref": "#/definitions/Omit<AAA,\\"b\\"_1>",
        },
      ],
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Partial_1 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "b": Object {
      "type": "string",
    },
    "c": Object {
      "type": "boolean",
    },
  },
  "type": "object",
}
`;

exports[`工具函数Partial_2 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "a",
        "b",
        "c",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "b": Object {
      "type": "string",
    },
    "c": Object {
      "$ref": "#/definitions/AAA",
    },
  },
  "type": "object",
}
`;

exports[`工具函数Partial_3 1`] = `
Object {
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "a",
        "b",
        "c",
      ],
      "type": "object",
    },
    "BBB": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "$ref": "#/definitions/AAA",
        },
      },
      "required": Array [
        "a",
        "c",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "a": Object {
      "type": "string",
    },
    "b": Object {
      "$ref": "#/definitions/BBB",
    },
  },
  "type": "object",
}
`;

exports[`工具函数Partial_联合类型_1 1`] = `
Object {
  "anyOf": Array [
    Object {
      "additionalProperties": false,
      "definitions": Object {
        "AAA": Object {
          "additionalProperties": false,
          "properties": Object {
            "a": Object {
              "type": "number",
            },
            "b": Object {
              "type": "string",
            },
            "c": Object {
              "type": "boolean",
            },
          },
          "required": Array [
            "a",
            "b",
            "c",
          ],
          "type": "object",
        },
      },
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "$ref": "#/definitions/AAA",
        },
      },
      "type": "object",
    },
    Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "type": "object",
    },
  ],
}
`;

exports[`工具函数Pick_1 1`] = `
Object {
  "properties": Object {
    "b": Object {
      "type": "string",
    },
  },
  "required": Array [
    "b",
  ],
  "type": "object",
}
`;

exports[`工具函数Pick_2 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "b": Object {
      "type": "string",
    },
  },
  "required": Array [
    "b",
  ],
  "type": "object",
}
`;

exports[`工具函数Pick_3 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "b": Object {
      "type": "string",
    },
  },
  "required": Array [
    "a",
    "b",
  ],
  "type": "object",
}
`;

exports[`工具函数Pick_3 2`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Pick<AAA,\\"b\\"|\\"c\\">": Object {
      "additionalProperties": false,
      "properties": Object {
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "b",
        "c",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Pick<AAA,\\"b\\"|\\"c\\">",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Pick_4 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Pick<AAA,\\"b\\"_0>": Object {
      "additionalProperties": false,
      "properties": Object {
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "b",
        "c",
      ],
      "type": "object",
    },
    "Pick<AAA,\\"b\\"_1>": Object {
      "additionalProperties": false,
      "properties": Object {
        "b": Object {
          "type": "string",
        },
      },
      "required": Array [
        "b",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "anyOf": Array [
        Object {
          "items": Object {
            "anyOf": Array [
              Object {
                "$ref": "#/definitions/Pick<AAA,\\"b\\"_0>",
              },
              Object {
                "$ref": "#/definitions/AAA",
              },
            ],
          },
          "type": "array",
        },
        Object {
          "$ref": "#/definitions/Pick<AAA,\\"b\\"_1>",
        },
      ],
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Record_1 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "about": Object {
      "type": "number",
    },
    "home": Object {
      "type": "number",
    },
  },
  "required": Array [
    "home",
    "about",
  ],
  "type": "object",
}
`;

exports[`工具函数Record_2 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "about": Object {
      "type": "string",
    },
    "home": Object {
      "type": "string",
    },
  },
  "required": Array [
    "home",
    "about",
  ],
  "type": "object",
}
`;

exports[`工具函数Record_3 1`] = `
Object {
  "additionalProperties": Object {
    "type": "string",
  },
  "properties": Object {
    "about": Object {
      "type": "string",
    },
    "contact": Object {
      "type": "string",
    },
    "home": Object {
      "type": "string",
    },
  },
  "required": Array [
    "home",
    "about",
    "contact",
  ],
  "type": "object",
}
`;

exports[`工具函数Record_4 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Record<\\"home\\"|\\"about\\"|\\"contact\\"|number,string>": Object {
      "additionalProperties": Object {
        "type": "string",
      },
      "properties": Object {
        "about": Object {
          "type": "string",
        },
        "contact": Object {
          "type": "string",
        },
        "home": Object {
          "type": "string",
        },
      },
      "required": Array [
        "home",
        "about",
        "contact",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Record<\\"home\\"|\\"about\\"|\\"contact\\"|number,string>",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Record_5 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Record<Page|number,string>": Object {
      "additionalProperties": Object {
        "type": "string",
      },
      "type": "object",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Record<Page|number,string>",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`工具函数Required_1 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "b": Object {
      "type": "string",
    },
    "c": Object {
      "type": "boolean",
    },
  },
  "required": Array [
    "a",
    "b",
    "c",
  ],
  "type": "object",
}
`;

exports[`工具函数Required_2 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "CCC": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "type": "object",
    },
  },
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "b": Object {
      "type": "string",
    },
    "c": Object {
      "$ref": "#/definitions/CCC",
    },
  },
  "required": Array [
    "a",
    "b",
    "c",
  ],
  "type": "object",
}
`;

exports[`工具函数Required_3 1`] = `
Object {
  "definitions": Object {
    "CCC": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "type": "object",
    },
    "DDD": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "$ref": "#/definitions/CCC",
        },
      },
      "required": Array [
        "a",
        "c",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "a": Object {
      "type": "string",
    },
    "b": Object {
      "$ref": "#/definitions/DDD",
    },
  },
  "required": Array [
    "a",
    "b",
  ],
  "type": "object",
}
`;

exports[`工具函数Required_联合类型_1 1`] = `
Object {
  "anyOf": Array [
    Object {
      "additionalProperties": false,
      "definitions": Object {
        "CCC": Object {
          "additionalProperties": false,
          "properties": Object {
            "a": Object {
              "type": "number",
            },
            "b": Object {
              "type": "string",
            },
            "c": Object {
              "type": "boolean",
            },
          },
          "type": "object",
        },
      },
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "$ref": "#/definitions/CCC",
        },
      },
      "required": Array [
        "a",
        "b",
        "c",
      ],
      "type": "object",
    },
    Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "number",
        },
        "b": Object {
          "type": "string",
        },
        "c": Object {
          "type": "boolean",
        },
      },
      "required": Array [
        "a",
        "b",
        "c",
      ],
      "type": "object",
    },
  ],
}
`;
